fix: Key store schema fixes

PhilWindle opened the pull request:
This PR modifies the keystore schema to use our native types and transformations where possible. It also fixes a potential discrepancy between the prover schema and the defined prover types.

spalladino commented on yarn-project/node-keystore/src/keystore_manager.ts:
```diff
@@ -265,7 +262,7 @@ export class KeystoreManager {
     const validator = this.getValidator(validatorIndex);
-    return validator.feeRecipient;
+    return validator.feeRecipient.toString();
   }
```
Would be nice to return an `AztecAddress` instead of plain string

PhilWindle replied:
Yep, done

spalladino reviewed (approved):
Thank you for removing all those casts 🙏

PhilWindle committed:
fix: Key store schema fixes

This PR modifies the keystore schema to use our native types and transformations where possible. It also fixes a potential discrepancy between the prover schema and the defined prover types.
